### PR TITLE
Add design doc for reference Cluster to Bundles

### PR DESCRIPTION
**Pull request by g-gaston**
Derived from [#1090](https://github.com/aws/eks-anywhere/issues/1090)
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: designs/cluster-bundles-reference.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# `Cluster` reference to `Bundles`
+
+## Introduction
+
+**Problem:** we rely on naming and a default namespace to find the `Bundles` object being used by an EKS-A `Cluster`.
+
+Currently, when we create a cluster, we rename the `Bundles` object we are using to have the same name as such `Cluster`. Then, when use that same name an same namespace to retrieve it from the cluster to rebuild a `cluster.Spec` in the CLI/controller.
```
Comment (jiayiwang7) [inline designs/cluster-bundles-reference.md:7]: is it my english or i can't understand this sentence

File: designs/cluster-bundles-reference.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+This is not a very clean pattern since it forces us to duplicate `Bundles` for multiple clusters. Also, in order to point a cluster to a new `Bundles` we need to update the existing one (or delete and recreate). Ideally, `Bundles` should be immutable in the cluster in the same way they are in real life.
```
Comment (jiayiwang7) [inline designs/cluster-bundles-reference.md:9]: should `Bundles` always be immutable? can eksa reconcile on bundle resources one day? i understand `Bundles` are more stateless CRD that eksa only reads it. But can user update certain component versions in bundle and expect it to work?

Comment (g-gaston) [reply designs/cluster-bundles-reference.md:9]: I would prefer if bundles were immutable.

Aside from making more sense from a CRD perspective, it makes cluster reconciliation easier: if you want to change a dependency, you need a new `Bundles` and point your `Cluster` to it.

That way we don't need to watch `Bundles` objects and back reference them to multiple `Cluster` objects when they change.

File: designs/cluster-bundles-reference.md (hunk 1 above)
Comment (g-gaston) [reply designs/cluster-bundles-reference.md]: Fixed :)

**Review by jiayiwang7 — APPROVED** [review]

**Pull request merged**